### ISSUE TITLE
-D no-copt implies foptimize = false

### ISSUE THREAD

**Issue by Simn** [HaxeFoundation/haxe#7550]
Which makes no sense

**Comment by Simn** [comment]
@ncannasse: Any idea what exactly `foptimize` should affect? Right now it only disables the `reduce_loop` filter, which is a pretty random thing to disable. Shouldn't it also imply `--no-inline` and disable the analyzer, and possibly other optimizations?

That's not strictly related to this issue but I'm still curious.

**Comment by ncannasse** [comment]
The idea of `--no-opt` is to disable optimizations and get in output code something quite the same as the code that was initially written by the developer

**Comment by Simn** [comment]
In that case we should either remove it or make it do that.

**Comment by ncannasse** [comment]
I agree with making it do that :) but that's not a top priority I think
